engine/verification/finder/engine.go

```diff
@@ -326,52 +355,202 @@ func (e *Engine) checkCachedReceipts() {
 				return
 			}
 
-			// checks if the result has already been handled
+			// checks if the result has already been processed or discarded
 			if e.processedResultIDs.Has(resultID) {
 				log.Debug().Msg("drops handling already processed result")
 				return
 			}
+			if e.discardedResultIDs.Has(resultID) {
+				log.Debug().Msg("drops handling already discarded result")
+				return
+			}
```

Comment:
Both these mempools will grow and begin ejecting items as we process/discard results. So it's possible that we receive a result that we have already either discarded or processed but its ID was ejected from the mempool. Do we handle this case?

Reply:
Per our conversation, we need to only keep track of a result status within a window of time when its receipts are floating in the network. Also, `match` engine is now filtering out receipts with block height lower than the last sealed block height. Created https://github.com/dapperlabs/flow-go/issues/5151 to move the logic from `match` to `finder` engine to have a more efficient pipeline.